seedlet/functions.php

```diff
@@ -150,22 +150,23 @@ function seedlet_setup() {
 		);
 
 		// Editor color palette.
-		$colors_theme_mod = get_theme_mod( 'custom_colors_active' );
-		$primary          = ( ! empty( $colors_theme_mod ) &&
-								'default' === $colors_theme_mod ||
-								empty( get_theme_mod( 'seedlet_--global--color-primary' ) ) ) ? '#000000' : get_theme_mod( 'seedlet_--global--color-primary' );
-		$secondary        = ( ! empty( $colors_theme_mod ) &&
-								'default' === $colors_theme_mod ||
-								empty( get_theme_mod( 'seedlet_--global--color-secondary' ) ) ) ? '#3C8067' : get_theme_mod( 'seedlet_--global--color-secondary' );
-		$foreground       = ( ! empty( $colors_theme_mod ) &&
-								'default' === $colors_theme_mod ||
-								empty( get_theme_mod( 'seedlet_--global--color-foreground' ) ) ) ? '#333333' : get_theme_mod( 'seedlet_--global--color-foreground' );
-		$tertiary = ( ! empty( $colors_theme_mod ) &&
-								'default' === $colors_theme_mod ||
-								empty( get_theme_mod( 'seedlet_--global--color-tertiary' ) ) ) ? '#FAFBF6' : get_theme_mod( 'seedlet_--global--color-tertiary' );
-		$background       = ( ! empty( $colors_theme_mod ) &&
-								'default' === $colors_theme_mod ||
-								empty( get_theme_mod( 'seedlet_--global--color-background' ) ) ) ? '#FFFFFF' : get_theme_mod( 'seedlet_--global--color-background' );
+		$wpcom_colors_array = get_theme_mod( 'colors_manager' );
+		if ( empty( $wpcom_colors_array ) ) {
+			// Use theme's built in color customization
+			$colors_theme_mod = get_theme_mod( 'custom_colors_active' );
+			$primary          = ( ! empty( $colors_theme_mod ) && 'default' === $colors_theme_mod || empty( get_theme_mod( 'seedlet_--global--color-primary' ) ) ) ? '#000000' : get_theme_mod( 'seedlet_--global--color-primary' );
+			$secondary        = ( ! empty( $colors_theme_mod ) && 'default' === $colors_theme_mod || empty( get_theme_mod( 'seedlet_--global--color-secondary' ) ) ) ? '#3C8067' : get_theme_mod( 'seedlet_--global--color-secondary' );
+			$foreground       = ( ! empty( $colors_theme_mod ) && 'default' === $colors_theme_mod || empty( get_theme_mod( 'seedlet_--global--color-foreground' ) ) ) ? '#333333' : get_theme_mod( 'seedlet_--global--color-foreground' );
+			$tertiary         = ( ! empty( $colors_theme_mod ) && 'default' === $colors_theme_mod || empty( get_theme_mod( 'seedlet_--global--color-tertiary' ) ) ) ? '#FAFBF6' : get_theme_mod( 'seedlet_--global--color-tertiary' );
+			$background       = ( ! empty( $colors_theme_mod ) && 'default' === $colors_theme_mod || empty( get_theme_mod( 'seedlet_--global--color-background' ) ) ) ? '#FFFFFF' : get_theme_mod( 'seedlet_--global--color-background' );
+		} else {
+			// Use wpcom color annotations
```

Comment:
If we go with this approach, these lines may need to be stripped out when we submit updates to the dotorg repo.

Reply:
Hmm yes, you're probably right. 🤔 That may get a little confusing. @allancole do you have any thoughts?

Reply:
Hmmm, I think we can roll all of this back to the wporg-specific method. Then, in `inc/wpcom.php` we can use `remove_theme_support()`([link](https://developer.wordpress.org/reference/functions/remove_theme_support/)) to remove _this_ method. Then later in that same file we can use `add_theme_support()` again to replace the wporg method with this color-annotation based, wpcom-specfic method.

Reply:
Not 100% sure that will work but, 🤞🏽

Reply:
I'm giving this a try, thanks for the suggestion @allancole!